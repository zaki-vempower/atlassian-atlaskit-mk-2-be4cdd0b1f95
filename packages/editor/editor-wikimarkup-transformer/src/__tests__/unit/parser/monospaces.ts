import WikiMarkupTransformer from '../../../index';

describe('JIRA wiki markup - Preformatted text', () => {
  const testCases: Array<[string, string]> = [
    [
      '[CS-1470] should NOT convert attachment links inside preformatted text',
      '{{[^test.txt]}}',
    ],
    [
      '[CS-1470] should NOT convert bold formatting inside preformatted text',
      '{{*bold*}}',
    ],
    [
      '[CS-1470] should NOT convert italic formatting inside preformatted text',
      '{{_italic_}}',
    ],
    [
      '[CS-1470] should convert raw link text inside preformatted text',
      '{{http://example.com}}',
    ],
    [
      '[CS-1470] should convert raw text around links inside preformatted text',
      '{{abc http://google.com abc}}',
    ],
    [
      '[CS-1470] should NOT convert citation inside preformatted text',
      '{{??citation??}}',
    ],
    [
      '[CS-1470] should NOT convert superscript inside preformatted text',
      '{{^superscript^}}',
    ],
    [
      '[CS-1470] should NOT convert strikethrough inside preformatted text',
      '{{-strikethrough-}}',
    ],
    [
      '[CS-1470] should NOT convert subscript inside preformatted text',
      '{{~subscript~}}',
    ],
    [
      '[CS-1470] should NOT convert rulers inside preformatted text',
      '{{-----}}',
    ],
    [
      '[CS-1553] should NOT convert bullet lists inside preformatted text',
      '{{* abc}}',
    ],
    [
      '[CS-1470] should convert color macro inside preformatted text',
      '{{{color:#59afe1}colour{color}}}',
    ],
    [
      '[CS-1470] should NOT convert wiki-links inside preformatted text',
      '{{[links|http://google.com]}}',
    ],
  ];

  for (const [testCaseDescription, markup] of testCases) {
    fit(testCaseDescription, () => {
      const transformer = new WikiMarkupTransformer();
      expect(transformer.parse(markup)).toMatchSnapshot();
    });
  }
});
