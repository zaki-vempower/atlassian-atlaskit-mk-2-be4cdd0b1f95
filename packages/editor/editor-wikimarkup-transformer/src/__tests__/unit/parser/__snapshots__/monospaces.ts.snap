// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert attachment links inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "[^test.txt]",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert bold formatting inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "*bold*",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert citation inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "??citation??",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert italic formatting inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "_italic_",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert rulers inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "-----",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert strikethrough inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "-strikethrough-",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert subscript inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "~subscript~",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert superscript inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "^superscript^",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should NOT convert wiki-links inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "[links|",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://google.com/",
              },
              "type": "link",
            },
          ],
          "text": "http://google.com",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "]",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should convert color macro inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "color": "#59afe1",
              },
              "type": "textColor",
            },
          ],
          "text": "colour",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should convert raw link text inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://example.com/",
              },
              "type": "link",
            },
          ],
          "text": "http://example.com",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1470] should convert raw text around links inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "abc ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://google.com/",
              },
              "type": "link",
            },
          ],
          "text": "http://google.com",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": " abc",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`JIRA wiki markup - Preformatted text [CS-1553] should NOT convert bullet lists inside preformatted text 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "type": "code",
            },
          ],
          "text": "* abc",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
